src/zenml/zen_server/routers/auth_endpoints.py

```diff
@@ -255,6 +258,10 @@ def generate_access_token(
     LOGIN,
     response_model=Union[OAuthTokenResponse, OAuthRedirectResponse],
 )
+@limiter.limit(
+    f"{server_config().login_rate_limit_minute}/minute;"
+    f"{server_config().login_rate_limit_day}/day"
+)
```

Comment:
I'm a bit worried about the fact that this rate limiting is counting all calls to this endpoint, even those that are legit. There might be cases in which logging with valid credentials will run into rate limiting errors. One example I can think of is many pipeline run steps running from the same location and logging in with service account API keys for example.

Reply:
I think there's a way to customize the conditions for rate-limiting. This GitHub issue might be useful: https://github.com/laurentS/slowapi/issues/31

Reply:
There are ways to customize that, cause I just set a key function to differentiate requests. The problem is that we have only the request at hand, not the response at the time of the decision.
I will merge this and do more research.

Reply:
@stefannica WDYT: https://github.com/zenml-io/zenml/pull/2484/commits/f6aaf7270b7c01bf8ad12d3e18e2832568099a7c